lib/collection/src/collection_manager/collection_updater.rs

```diff
@@ -19,11 +19,9 @@ impl CollectionUpdater {
         match operation_result {
             Ok(_) => {
                 if !segments.read().failed_operation.is_empty() {
+                    // If this operation failed before, remove it because it got fixed now
                     let mut write_segments = segments.write();
-                    if write_segments.failed_operation.contains(&op_num) {
-                        // Failed operation successfully fixed
-                        write_segments.failed_operation.remove(&op_num);
-                    }
+                    write_segments.failed_operation.remove(&op_num);
```

Comment:
Nitpicking but maybe call it something like `segments_write_guard`. But upto you :)

Reply:
I removed the intermediate write guard entirely, so no naming is needed anymore :+1: